File: app/src/main/java/columner/MyGLSurfaceView.java
package columner;

import android.content.Context;
import android.opengl.GLSurfaceView;
import android.util.DisplayMetrics;
import android.view.MotionEvent;

public class MyGLSurfaceView extends GLSurfaceView {
    private MyGLRenderer mRenderer;
    private Context context;
    private DisplayMetrics displayMetrics;
    private int width;
    private float x;

    public MyGLSurfaceView(final Context context) {
        super(context);

        this.context = context;

        // Create an OpenGL ES 2.0 context
        this.setEGLContextClientVersion(2);

        mRenderer = new MyGLRenderer(this.context);

        // Set the Renderer for drawing on the GLSurfaceView
        this.setRenderer(mRenderer);
    }

    @Override
    public boolean onTouchEvent(final MotionEvent event) {
        if (event.getEventTime() - event.getDownTime() < 1) {
            displayMetrics = context.getResources().getDisplayMetrics();
            width = displayMetrics.widthPixels;

            x = event.getX();

            if (!mRenderer.falling) {
                if (x <= width / 3) {
                    mRenderer.nextCol = 0;
                } else if (x > width / 3 && x < (width << 1) / 3) {
                    mRenderer.nextCol = 1;
                } else if (x > (width << 1) / 3) {
                    mRenderer.nextCol = 2;
                }
            }

            if (!mRenderer.started) {
                mRenderer.started = true;
            }

            if (!mRenderer.jump && !mRenderer.falling) {
                mRenderer.jump = true;
            }

            this.requestRender();
        }

        return true;
    }
}

File: app/src/main/java/columner/MyGLSurfaceView.kt
package columner

import android.content.Context
import android.opengl.GLSurfaceView
import android.view.MotionEvent

class MyGLSurfaceView(context: Context) : GLSurfaceView(context) {
    private val mRenderer: MyGLRenderer
    private val surfaceWidth = context.getResources().getDisplayMetrics().widthPixels

    init {
        this.setEGLContextClientVersion(2)
        mRenderer = MyGLRenderer(context)

        this.setRenderer(mRenderer)
    }

    override fun onTouchEvent(event: MotionEvent): Boolean {
        if (event.getEventTime() - event.getDownTime() < 1) {
            val x = event.getX() * 3

            if (!mRenderer.falling) {
                if (x <= surfaceWidth) {
                    mRenderer.nextCol = 0
                } else if (x > surfaceWidth && x < surfaceWidth shl 1) {
                    mRenderer.nextCol = 1
                } else if (x > surfaceWidth shl 1) {
                    mRenderer.nextCol = 2
                }
            }

            mRenderer.started = true
            mRenderer.jump = !mRenderer.jump && !mRenderer.falling;

            this.requestRender()
        }

        return true
    }
}